ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -365,27 +364,38 @@ def _clear_instance_recordings(self, instance: Any) -> None:
         if hasattr(instance, "_datadog_spans"):
             delattr(instance, "_datadog_spans")
 
-    def _llmobs_set_metadata(self, span: Span, model_provider: Optional[str] = None) -> None:
-        if not model_provider:
+    def _llmobs_set_metadata(self, span: Span, kwargs: Dict[str, Any]) -> None:
+        identifying_params = kwargs.pop("_dd.identifying_params", None)
+        if not identifying_params:
             return
+        metadata: Dict[str, Any] = {}
+        for _, val in identifying_params.items():
+            if isinstance(val, dict):
+                params = self._llmobs_extract_parameters(val)
+            else:
+                params = self._llmobs_extract_parameters(identifying_params)
+            if params["max_tokens"] is not None and params["max_tokens"] != "None":
+                metadata["max_tokens"] = int(params["max_tokens"])
+            if params["temperature"] is not None and params["temperature"] != "None":
+                metadata["temperature"] = float(params["temperature"])
 
-        metadata = {}
-        temperature = span.get_tag(f"langchain.request.{model_provider}.parameters.temperature") or span.get_tag(
-            f"langchain.request.{model_provider}.parameters.model_kwargs.temperature"
-        )  # huggingface
-        max_tokens = (
-            span.get_tag(f"langchain.request.{model_provider}.parameters.max_tokens")
-            or span.get_tag(f"langchain.request.{model_provider}.parameters.maxTokens")  # ai21
-            or span.get_tag(f"langchain.request.{model_provider}.parameters.model_kwargs.max_tokens")  # huggingface
-        )
-
-        if temperature is not None and temperature != "None":
-            metadata["temperature"] = float(temperature)
-        if max_tokens is not None and max_tokens != "None":
-            metadata["max_tokens"] = int(max_tokens)
         if metadata:
             span._set_ctx_item(METADATA, metadata)
 
+    def _llmobs_extract_parameters(self, parameters: Dict[str, Any]) -> Dict[str, Any]:
+        metadata: Dict[str, Any] = {}
+        max_tokens = None
+        temperature = None
+        for k, v in parameters.items():
+            if k == "temperature":
+                temperature = v
+            elif k in ["max_tokens", "maxTokens", "max_completion_tokens"]:
+                max_tokens = v
+        metadata["max_tokens"] = max_tokens
+        metadata["temperature"] = temperature
```

Comment:
```suggestion
     temperature = parameters.get("temperature")
     if temperature is not None or temperature is not "None":
         metadata["temperature"] = float(temperature)
     for key in ("max_tokens", "maxTokens", "max_completion_tokens"):
         max_tokens = parameters.get(key)
         if max_tokens is not None or max_tokens is not "None":
             metadata["max_tokens"] = int(max_tokens)
             break
```
Maybe iterating through all params to find two kv pairs is excessive